### DISCUSSION TITLE
I want to read all .hbs files and render them

### DISCUSSION THREAD

**Discussion by vanhoutenbos** [Handlebars-Net/Handlebars.Net#548] [General]
Hi,

I am a bit confused, normally in handlebars (using javascript etc.) I can just let the views be rendered based on the standard folder structure.
Here it seems like I need to copy & paste all HTML to the C# code which seems a bit much so I must be missing something!

I am wondering how I can read all source and templates from the image below and render them as HTML

![image](https://user-images.githubusercontent.com/10237926/236458086-e9619f71-2d67-4bbf-bbd3-ddc98070496b.png)

**Comment by rexm** [comment]
>normally in handlebars (using javascript etc.) I can just let the views be rendered based on the standard folder structure

I don't think that's right. Handlebars doesn't have any notion of files or folders. If you check out the [handlebars documentation](https://handlebarsjs.com/) you won't see any reference to views or folders. There are frameworks and view engines that _use_ handlebars as their templating language - the experience you're describing, you must have been using one of those. There are some equivalent view engines out there for c# that use this library as their core engine. Without knowing more about your application architecture or requirements, I can't recommend a specific one, but hopefully that gets you started.

**Reply by vanhoutenbos** [reply]
Thank you Rex! That sounds like that it what I was using before.

Maybe you can advice me on which library to use.
We are creating an azure function that generates a resume based on JSON Resume.
I tried porting their cli to a azure function but without success so I am planning on using handlebars + headless puppeteer for a pdf generated resume.
The pages, css and images are static only the JSON is variable.

**Reply by rexm** [reply]
I don't know about that. If I were you I'd probably use [embedded resources](https://learn.microsoft.com/en-us/dotnet/core/extensions/create-resource-files) to let me keep the templates as individual files in source but compile them into the application as strings.

**Reply by vanhoutenbos** [reply]
Thank you for your suggestions, I will investigate!
